src/form/Form.js

```diff
@@ -0,0 +1,170 @@
+                new Input()
+                    .id("number-test")
+                    .label("Number Test")
+                    .type("number")
+                    .validate("\\d+")
+                    .value(123),
+                new Input()
+                    .id("checkbox-test")
+                    .label("Checkbox Test")
+                    .type("checkbox")
+                    .value(true),
+                new Input()
+                    .id("select-test")
+                    .label("Select Test")
+                    .type("select")
+                    .selectOptions(['A','B','C'])
+                    .value('B'),
+                new Input()
+                    .id("button-test")
+                    .label("Button Test")
+                    .type("button")
+                    .value('Button Text'),
+                new Input()
+                    .id("button-test")
+                    .label("Textarea Test")
+                    .type("textarea")
+                    .value('Textarea Text')
+            ])
+        ;
+        var context = this;
+        this
+            .controls([
+                new Input()
+                    .id("submit-btn")
+                    .type("button")
+                    .value("Submit")
+                    .click(function(){
+                        context.submit();
+                    }),
+                new Input()
+                    .id("clear-btn")
+                    .type("button")
+                    .value("Clear")
+                    .click(function(){
+                        context.clear();
+                    })
+            ]);
+        return this;
+    };
+    
+    Form.prototype.submit = function(){
+        var isValid = true;
+        if(this.validate()){
+            isValid = this.checkValidation();
+        }
+        if(isValid){
+            var dataArr = [];
+            var inpArr = this.inputs();
+            inpArr.forEach(function(inp){
+                dataArr.push({id:inp.id(),value:inp.value()});
+            });
+            console.log('Clicked Submit: '+JSON.stringify(dataArr));
+        }
+    };
+    Form.prototype.clear = function(){
+        var inpArr = this.inputs();
+        inpArr.forEach(function(inp){
+            if(inp.type() === "checkbox"){
+                inp.value(false);
+                inp._inputElement.node().checked = false;
+            } else {
+                inp.value(' ');
+                inp._inputElement.node().value = ' ';
+            }
+        });
+    };
+    
+    Form.prototype.checkValidation = function(){
+        var ret = true;
+        var msgArr = [];
+        this.inputs().forEach(function(inp){
+            if(!inp.isValid()){
+                msgArr.push("'"+inp.label()+"'"+' value is invalid.');
+            }
+        });
+        if(msgArr.length > 0){
+            alert(msgArr.join('\n'));
+            ret = false;
+        }
+        return ret;
+    };
```

Comment:
look into firing an event? and being able to pass this data with it so that end user can customize the way the popup looks